packages/components/src/DatePicker/subcomponents/DateInputField/DateInputField.spec.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useRef } from "react"
+import { render, screen } from "@testing-library/react"
+import userEvent from "@testing-library/user-event"
+import { enUS } from "date-fns/locale"
+import { DateInputField, DateInputFieldProps } from "./DateInputField"
+
+const user = userEvent.setup()
+
+const defaultProps: DateInputFieldProps = {
+  id: "test__date-input-field",
+  labelText: "Bacon expiry",
+  onButtonClick: jest.fn<void, []>(),
+  onKeyDown: jest.fn<void, [React.KeyboardEvent<HTMLInputElement>]>(),
+  value: undefined,
+  locale: enUS,
+}
+
+const DateInputFieldWrapper = (
+  props: Partial<DateInputFieldProps>
+): JSX.Element => <DateInputField {...defaultProps} {...props} />
+
+describe("<DateInputField />", () => {
+  describe("Input", () => {
+    it("associates the description with the input", () => {
+      render(<DateInputFieldWrapper />)
+      expect(
+        screen.getByRole("textbox", {
```

Comment:
Why use `screen` here instead of getting it from the `render`? I've been grabbing from the `render`...

Reply:
Because I didn't write this 😄 I always grab from `render` too.
I did look into it, and it seems that `screen` is more of an alias which makes transitions from other testing libraries (eg. Cypress) a little easier.

Did a tiny bit of testing with an existing test:
- Using `screen` - Cached time `1.917 s`
- Using `render` - Cached time `1.8 s`

It seems `render` is consistently slightly more performant (which makes sense, given it can skip the step of searching for the container). Happy to change it.